src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py

```diff
@@ -81,8 +83,9 @@ def _main(settings, optlist, args):
         else:
             log.error("Error: No exception thrown, but XUnit results not created")
             test_executor.report_error(settings, failure_type="XUnitTestFailure")
+            return WORKITEM_FAILED
```

Comment:
I can log the return code prior to that in case in does in fact lend some meaning. I have noticed that if xunit fails to run correctly though we still get a random number on occasion.